### DISCUSSION TITLE
HERON environment installation, pyomo issue

### DISCUSSION THREAD

**Discussion by glher** [idaholab/HERON#101] [Q&A] [answered]
I was trying to generate the documentation for HERON, and the conda environment had trouble resolving the pyomo dependency.

Steps taken:

    $ git clone https://github.com/idaholab/raven.git
    $ cd raven/scripts
    $ ./establish_conda_env.sh --install --conda-defs /Users/glher/opt/miniconda3/etc/profile.d/conda.sh
      [...]
      ... done!
    $ conda activate raven_libraries
    $ ./build_raven
      ... done!
    $ ./run_tests -j2
      [...]
      ... tests passed!
    $ cd ../..
    $ git clone https://github.com/idaholab/heron.git
    $ cd heron/doc
    $ ./make_docs.sh
      IOError: HERON config file not found at "/Users/glher/Projects/INL/Codes/heron/.ravenconfig.xml"! Has HERON been installed as a plugin in a RAVEN installation?
    make: *** [populate] Error 1
      ... ... failed to make docs in user_manual 
    $ cd ../../raven/scripts
    $ ./install_plugins.py -s /Users/glher/Projects/INL/Codes/heron
      [...]
      ... plugin "heron" succesfully installed!
    $ cd ../../heron/doc
    $ ./make_docs.sh
      ModuleNotFoundError: No module named 'pyomo'
    $ conda install pyomo=5.7.0
      PackagesNotFoundError: The following packages are not available from current channels:
      - pyomo=5.7.0
    $ cd user_manual/
    $ make
      ModuleNotFoundError: No module named 'pyomo' 
    $ conda install -c conda-forge pyomo=5.7.0
      Executing transaction: done
    $ make
      [...]
      Output written on HERON_user_manual.pdf (42 pages, 406804 bytes).

I have never really used conda, I've always been a pip guy, so it probably comes from my misunderstanding of what conda does, or my misunderstanding (for now) of the HERON-RAVEN communication.

What did I do wrong here, and how could I have had less issues installing the correct environment?

**Comment by dylanjm** [comment, marked as answer]
Can you try the following commands and see if this fixes your error?

```
$ cd raven
$ ./scripts/install_plugins.py -s <your/path/to/HERON>
$ ./scripts/establish_conda_env.sh --install
$ ./build_raven
$ ./run_tests -j 6
$ cd ../HERON/doc
$ ./make_docs.sh
```

Typically you have to install the HERON plugin first and then setup the conda env. This will install packages like 'pyomo' and should fix that error.

**Reply by glher** [reply]
Ah, I'm dumb.

I had forgotten to run the establish_conda_env with the install flag, so it ignored the HERON plugin addition when resolving the environment.

Thanks!
